simconvoi.h

```diff
@@ -495,7 +495,23 @@ class convoi_t : public sync_steppable, public overtaker_t
 	/// Pushes the convoy stopping time at the current halt to the schedule
 	void push_convoy_stopping_time();
 
+
+
+
+
 public:
+	// Reverses the order of the convoy.
+	// @author: jamespetts
+	bool reversing;// Whether this convoy's vehicles are currently arranged in reverse order. The flag for calculation of position.
+	bool is_reversed() const;
+	// Reorder the vehicle array
+	// Can be executed even with a vehicle array that does not belong to convoy for UI
+	void reverse_vehicles_at_halt();
+	void reverse_vehicles_while_driving();
+	void reverse_vehicles_go_depot();
```

Comment:
TODO: The public function should be merged. The individual logic should be moved to private.